main.go

```diff
@@ -32,13 +32,19 @@ func main() {
 		return
 	}
 
-	opts := options.NewOptions()
-	err := options.Load(*config, flagSet, opts)
+	legacyOpts := options.NewLegacyOptions()
```

Comment:
Should the subset of options that are now in legacy_options be contained in the main `options.Options`? Seems a little backwards that the LegacyOptions holds all the `options.Options` inside of it instead of the reverse.

Reply:
So I chose to do it this way round because I believe it has a few benefits:
- When you load a LegacyOptions, you have to pass it through conversion before passing it into the OAuth2 Proxy, guaranteeing conversion
- `Options` this way, has only the options that are required for running the OAuth2 Proxy, none of the legacy options are accessible so it forces us to do the migrations within the code, otherwise we could change a legacy option in the code and just move the location to `options.Legacy.Thing`
- When we start getting closer to the end of this project, we can use the Legacy Load functionality with LegacyOptions to Options conversion as a separate binary to help people migrate between versions, I think this would be harder or weirder if we embedded the opposite way around
- As we get closer to the end of this and start testing the structural schema, we won't have the extra legacy fields as options within the YAML structure which could confuse things if we set those accidentally

WDYT?

Reply:
Options 1 & 2 sold me -- sounds good to me!

Forces us to be more thoughtful about the design of the structs & methods that configurations (legacy or future structured) lead to.

Reply:
Which reminds me -- is there in issue where we can start brainstorming and dumping what a conversion to a structured nested config would look like for all of the existing flags?

A lot lend themselves to pretty self explanatory groupings already (logging, cookies, sessions, provider specifics) - but other singleton settings probably need more thought & discussion.

Reply:
There's an issue about the RFC, we could discuss it there, or maybe create a separate one for the design that we can keep up to date?

Reply:
Using the RFC makes a lot of sense.  Should we pin that issue?

Reply:
I believe it already is pinned 😜

Reply:
It is now pinned